HW2/Task2.2/Program.cs

```diff
@@ -0,0 +1,121 @@
+﻿// TODO  I have no idea what means indexers in the Task 2.2.
+
+DiagonalMatrix matrix1 = new DiagonalMatrix(1, 2, 3);
+DiagonalMatrix matrix2 = new DiagonalMatrix(1, 2, 3);
+
+Console.WriteLine(matrix1.ToString());
+
+int track = matrix1.Track();
+Console.WriteLine($"Sum of the matrix elements is: {track}");
+
+bool isEqual = matrix1.Equals(matrix2);
+Console.WriteLine($"If {nameof(matrix1)} is equal to {nameof(matrix2)}?: {isEqual}");
+
+public class DiagonalMatrix
+{
+    public int Size { get; init; }
+
+    private int[] valuesOnDiagonal;
+
```

Comment:
not needed, extra memory, not effective. Use _valuesOnDiagonal only for all operations, remove readyMatrix